test/helper/e2e2/operator/embedded.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2025 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operator
+
+import (
+	"context"
+	"flag"
+	"os"
+	"strconv"
+	"sync"
+)
+
+func RunEmbeddedSet() bool {
+	envSet, _ := strconv.ParseBool(os.Getenv("RUN_EMBEDDED"))
+	return envSet
+}
+
+type RunnerFunc func(context.Context, *flag.FlagSet, []string) error
+
+type EmbeddedOperator struct {
+	runnerFunc RunnerFunc
+	mutex      sync.Mutex
+	wg         sync.WaitGroup
+	ctx        context.Context
+	cancelFn   context.CancelFunc
+	args       []string
+}
+
+func NewEmbeddedOperator(runnerFunc RunnerFunc, args []string) *EmbeddedOperator {
+	return &EmbeddedOperator{runnerFunc: runnerFunc, args: args}
+}
+
+func (e *EmbeddedOperator) Start(t testingT) {
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
+	t.Logf("starting operator in process with args: %v", e.args)
+
+	if e.ctx != nil {
+		return
+	}
+	e.ctx, e.cancelFn = context.WithCancel(context.Background())
+	e.wg.Add(1)
+	go func() {
+		defer e.wg.Done()
+		fs := flag.NewFlagSet("direct-ako-run", flag.ContinueOnError)
```

Comment:
is `direct-ako-run` actually used?

Reply:
> NewFlagSet returns a new, empty flag set with the specified name and error handling property. **If the name is not empty, it will be printed in the default usage message and in error messages.**

I guess we are not using as we are not interacting with the flags help on tests. Maybe I can pass empty string for the embedded run instead.